cmd/cli/kubectl-kyverno/processor/policy_processor.go

```diff
@@ -269,7 +269,8 @@ func (p *PolicyProcessor) makePolicyContext(
 	policyContext = policyContext.
 		WithPolicy(policy).
 		WithNamespaceLabels(namespaceLabels).
-		WithResourceKind(gvk, subresource)
+		WithResourceKind(gvk, subresource).
+		WithAdmissionOperation(true)
```

Comment:
If i'm not mistaken the cli never receives an admission operation 🤔

Reply:
The bug is reached when the flag `mutateExistingOnPolicyUpdate: true` is set on a resource and you try to run kyverno test/ kyverno apply on a resource with that kyverno Policy.

I saw that you can set the operation in the Values file of a kyverno test but that didn't fix the issue.
Then i looked into the code and i reasoned that `AdmissionOperation` is the case where `request.operation` is any of `[Create, Update, Delete]`. 

While a background operation to refresh existing resources would be 'Background'.

Without this change you end up in the mutate branch, which causes kyverno to do api calls using a kubernetes client, which is set to `nil` because it is in CLI mode.
```
if !policyContext.AdmissionOperation() && rule.HasMutateExisting() {
```